src/net_processing.cpp

```diff
@@ -2936,9 +2937,11 @@ void ProcessMessage(
 
         TxValidationState state;
 
-        nodestate->m_tx_download.m_tx_announced.erase(hash);
-        nodestate->m_tx_download.m_tx_in_flight.erase(hash);
-        EraseTxRequest(hash);
+        for (const GenTxid& gtxid : {GenTxid(false, txid), GenTxid(true, wtxid)}) {
```

Comment:
I agree with @naumenkogs (https://github.com/bitcoin/bitcoin/pull/19569#discussion_r460714058) that this change should be introduced in _p2p: enable fetching of orphans from wtxid peers_ instead of _refactor: add GenTxid (=txid or wtxid) type and use it for tx request logic_

Reply:
Done.